### DISCUSSION TITLE
pong not being sent

### DISCUSSION THREAD

**Discussion by NickFritz1982** [uNetworking/uWebSockets.js#408] [Q&A] [answered]
In the doc's it says that ping/pong is optional as it automatically send a pong when a ping comes in.

`Handler for received ping control message. You do not need to handle this, pong messages are automatically sent as per the standard.`

This isn't happening so I've added the following code but it still isn't sending a pong response?

Am I doing something wrong?

ping: (ws) => {
      ws.pong();
      ws.send(0xA);
      console.log(`ping()... 0xA sent back`);
      
    },

Running version 18.8

**Comment by hst-m** [comment, marked as answer]
I just tried this with [ws](https://www.npmjs.com/package/ws) client library and ping/pong works as expected, pong automatically sent back to client. You did not specify what client you are using to send Ping to server. What client are you using and what code is the client sending Ping with? Is your console.log showing that the ping event gets called on the server?
